### ISSUE TITLE
Very long black screen when starting the android app exported from Unity

### ISSUE THREAD

**Issue by amitarora456** [artoolkit/arunity5#119]
Hi,

I am using Unity 5.5 and ArToolKit 5.3.2,  I have created a project which has 30 models which are kepts in Assets folder. I have reduced the size of the texture if all the models but when I build in Android, the app takes too long to show the camera. i.e a very long black screen appears.  The black screen stays for about 40 seconds.  Am I missing something which made it so slow?

**Comment by thanuras** [comment]
Yep, I got the same issue with our AR children's book, we just submitted the iOS build to App store for review. But we really want this issue to be fixed as it is not a good experience for the user. I'm using Unity 5.3. and ARToolkit 5.3.2. 
I can see this is the time it takes to load the points (at leas that what debug screen says). I have 14 NFT markers on my app, and they've been generated using following parameters for genTexData
Select extraction level for tracking features : 1
Select extraction level for initializing features:0
Minimum image resolution:50
Maximum image resolution:150
This is on 300 dpi image.
Initial app load time on an iPad 3 (Target user device with the lowest config) takes 50sec - 60 sec, then the black screen time is over a minute. It loads 168 points (which I really don't know what exactly the points are)
Is there any way that we can reduce this time? Is there any specific way to generate markers to reduce this time?
Currently I have an image that says "wait until we load app resources..." but it's there for a long time, really like to fix this one. 
Value your help..

**Comment by Cambesa** [comment]
I have almost the same issue(Galaxy S5 Mini): In my case the black screen that stays for a long time before the "Made with Unity" screen appears, also shows a square in the top-left which is streaming the camera recording. Aside from this problem I also have an extra one: Most of the time, the "Made with Unity" screen is replaced by junk data/memory like: https://pbs.twimg.com/media/CzeK0CsXgAEH8Ml.jpg Does someone have any tips for me to get rid of this bug? I have looked for buffers in the code but can not find a video buffer

**Comment by namkazt** [comment]
Here is my fix for camera problem relate to loading app time and corner camera small box.
https://drive.google.com/open?id=0B5vFwC-dOjOpckk4TXB3d0FXS3c

How to do it:
1, Replace UnityARPlayer.jar with your current using.
2, Add ARToolkitHelper to any where you want to place
3, Should uncheck Auto start ARToolkit in ARController and call it in some place you want to start AR Camera and using this code: 
```
ARController arController = GameObject.Find("ARToolKit").GetComponent<ARController>();
            if (Application.platform == RuntimePlatform.Android)
            {
                // AR helper is instance of ARToolkitHelper
               // just use  _ARHelper = new ARToolkitHelper();
                _ARHelper.OpenCamera();
            }
            arController.StartAR();
```
There is some funcs:
1, OpenCamera
2, CloseCamera
3, PauseCamera ( it just stop send frame buffer data from plugin to ARtoolkit to void process. i use it when open other menu that don't need camera process any more  )
4, ResumeCamera

thank me later.

**Comment by namkazt** [comment]
#130  this PR will fix for bug. 
Long loading time relate to what marker you use ( maybe  ?? ) i was testing if i'm using NFT with too many features it will take a super long time to load. so the solution here is. add Marker when do you need it. that mean we load all ARConfig, start camera. then create a loading scene or something to load Marker.

**Comment by AnkitAndroaid** [comment]
I downloaded the android sample and tried to run it. It takes more than a minute to recognise the marker for the first time. After that it detects the marker easily. Why is it so? And there is A Black screen for around 30 Seconds when i Run the NFT sample app. Could not find the solution to minimise these delays and make it a better UX for users.

**Comment by 5y** [comment]
@namkazt Thank you for reply. Could you please share more detailed solution?

**Comment by quangtn91** [comment]
@5y  see #68

**Comment by 5y** [comment]
@quangtn91  Yes, this problem is related to the checking calibration parameters from ARtoolkit server[1]. However, I would like to disable this function and put the default Calibration file. I found some solutions for Artoolkit android[2]. but still nothing for Unity. is there any idea ?
Thanks.

1- https://github.com/artoolkit/arunity5/issues/68
2- https://github.com/artoolkit/artoolkit5/issues/347#issuecomment-338500605

**Comment by josemcasasp** [comment]
I found a solution that works for me.

First i get from the repo github the late version of the unitypackage in my project.
Then in Plugins/Android/libs/armeabi-v7a delete the libc++_shared file.

Maybe i did something else that i forgotten but dont think so.

**Comment by ebruekurnaz** [comment]
> I downloaded the android sample and tried to run it. It takes more than a minute to recognise the marker for the first time. After that it detects the marker easily. Why is it so? And there is A Black screen for around 30 Seconds when i Run the NFT sample app. Could not find the solution to minimise these delays and make it a better UX for users.

Same issue, Did you find any solution to that ?
